elf_reader_test.go

```diff
@@ -474,6 +486,41 @@ func TestLibBPFCompat(t *testing.T) {
 	})
 }
 
+func loadTargetProgram(tb testing.TB, name string, opts CollectionOptions) (*Program, *Collection) {
+	file := "test_pkt_access.o"
+	program := "test_pkt_access"
+	switch name {
+	case "new_connect_v4_prog":
+		file = "connect4_prog.o"
+		program = "connect_v4_prog"
+	case "new_do_bind":
+		file = "connect4_prog.o"
+		program = "connect_v4_prog"
+	case "freplace_cls_redirect_test":
+		file = "test_cls_redirect.o"
+		program = "cls_redirect"
+	case "new_handle_kprobe":
+		file = "test_attach_probe.o"
+		program = "handle_kprobe"
+	case "test_pkt_md_access_new":
+		file = "test_pkt_md_access.o"
+		program = "test_pkt_md_access"
+	default:
```

Comment:
Most of the freplace selftests use test_pkt_access from test_pkt_access.o as their attachment point. I made that the default case to avoid having to list every self test (see https://github.com/cilium/ebpf/pull/288/files#diff-cd933a7ae9404685007326714b90c84730de14ecbd8ad1380cbfbb3e2a9ed0a8R490). Will add a comment here to avoid confusion.